gobblin-core/src/main/java/gobblin/writer/AvroHdfsTimePartitionedWriter.java

```diff
@@ -187,6 +177,21 @@ public void write(GenericRecord record) throws IOException {
     this.pathToWriterMap.get(writerOutputPath).write(record);
   }
 
+  /**
+   * Checks whether the given value is present and can be cast to Long.
+   */
+  private boolean isPartitionValueUsable(Optional<Object> writerPartitionColumnValue) {
+    if (!writerPartitionColumnValue.isPresent()) {
+      return false;
+    }
+    try {
+      Long.parseLong(writerPartitionColumnValue.get().toString());
```

Comment:
You can use `Longs.tryParse`, http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/primitives/Longs.html#tryParse(java.lang.String).